src/google/protobuf/util/field_mask_util.h

```diff
@@ -46,6 +46,12 @@ namespace google {
 namespace protobuf {
 namespace util {
 
+#if GTEST_HAS_RTTI
```

Comment:
Could you do `#if PROTOBUF_RTTI` instead? I assume the gtest macro will work in tests but not for real production binaries. Could you also make sure to undefine the macro after it's used, just to make sure it can't conflict with other macros? It might also be a good idea to call it `PROTOBUF_TYPENAME()` just to be safe.

Reply:
Thank you for your comments.

I've cleaned up the patch as per your suggestions.